### PR TITLE
Including brokerage fees in Money Out calculation (if not ignoring them) - Reopened

### PR DESCRIPTION
**Pull request by fabrica64**
Including brokerage fees in Money Out calculation (if not ignoring them) in advanced-portfolio.scm
I reopened original PR #1647 because I inadvertently closed it during a fork sync...

### CODE REVIEW

**Comment by dawansv** [comment]
@fabrica64 Can you also update gnucash/report/reports/standard/test/test-portfolios.scm ? The "advanced" tests need to be updated with the new money out value. The test data has a $10 selling fee, so it needs to be $543.00 instead of $553.00.

See the failed test results below:
```
98: [pass] line:64, test: null-test portfolio
98: [pass] line:64, test: null-test advanced-portfolio
98: [pass] line:73, test: portfolio: pricedb-latest
98: [pass] line:80, test: portfolio: pricedb-nearest
98: [pass] line:87, test: portfolio: average-cost
98: [pass] line:93, test: portfolio: weighted-average
98: [fail] line:104, test: advanced: average basis
98: test-portfolios.scm/advanced-portfolio-tests
98:  -> expected: ("AAPL" "AAPL" "NASDAQ" "42.00" "$6.0000" "$484.88" "$252.00" "$800.00" "$553.00" "$227.88" "-$232.88" "-$5.00" "-0.63%" "$4.00" "$10.00" "-$1.00" "-0.13%")
98:  -> obtained: ("AAPL" "AAPL" "NASDAQ" "42.00" "$6.0000" "$484.88" "$252.00" "$800.00" "$543.00" "$227.88" "-$232.88" "-$5.00" "-0.63%" "$4.00" "$10.00" "-$1.00" "-0.13%")
98: [fail] line:112, test: advanced: fifo basis
98: test-portfolios.scm/advanced-portfolio-tests
98:  -> expected: ("AAPL" "AAPL" "NASDAQ" "42.00" "$6.0000" "$543.94" "$252.00" "$800.00" "$553.00" "$286.94" "-$291.94" "-$5.00" "-0.63%" "$4.00" "$10.00" "-$1.00" "-0.13%")
98:  -> obtained: ("AAPL" "AAPL" "NASDAQ" "42.00" "$6.0000" "$543.94" "$252.00" "$800.00" "$543.00" "$286.94" "-$291.94" "-$5.00" "-0.63%" "$4.00" "$10.00" "-$1.00" "-0.13%")
98: [fail] line:120, test: advanced: filo basis
98: test-portfolios.scm/advanced-portfolio-tests
98:  -> expected: ("AAPL" "AAPL" "NASDAQ" "42.00" "$6.0000" "$400.00" "$252.00" "$800.00" "$553.00" "$143.00" "-$148.00" "-$5.00" "-0.63%" "$4.00" "$10.00" "-$1.00" "-0.13%")
98:  -> obtained: ("AAPL" "AAPL" "NASDAQ" "42.00" "$6.0000" "$400.00" "$252.00" "$800.00" "$543.00" "$143.00" "-$148.00" "-$5.00" "-0.63%" "$4.00" "$10.00" "-$1.00" "-0.13%")
98: Source:test-portfolios.scm
98: pass = 6, fail = 3
```

This is the section of test-portfolios.scm that needs to be updated
https://github.com/Gnucash/gnucash/blob/d214b2f05d014169307a02e523ba4cfb8875309e/gnucash/report/reports/standard/test/test-portfolios.scm#L98-L125

**Comment by fabrica64** [comment]
> @fabrica64 Can you also update gnucash/report/reports/standard/test/test-portfolios.scm ? The "advanced" tests need to be updated with the new money out value. The test data has a $10 selling fee, so it needs to be $543.00 instead of $553.00.

Tests updated.

**Comment by dawansv** [comment]
@fabrica64 see https://wiki.gnucash.org/wiki/An_Introduction_to_Git#Pull_Requests if more info is needed on how to sync your stable branch with the main gnucash repository and rebase your other branches if needed. It shouldn't take the form of an additional commit, however.
Also see on that same page how in the future it is better to create another branch for your changes, not use your copy of the stable branch.

**Comment by fabrica64** [comment]
> @fabrica64 see https://wiki.gnucash.org/wiki/An_Introduction_to_Git#Pull_Requests if more info is needed on how to sync your stable branch with the main gnucash repository and rebase your other branches if needed. It shouldn't take the form of an additional commit, however. Also see on that same page how in the future it is better to create another branch for your changes, not use your copy of the stable branch.

Ok, understood. How can I fix this now? Should I close the pull request and reopen it?

File: gnucash/report/reports/standard/advanced-portfolio.scm [hunk 1]
```diff
@@ -129,7 +129,7 @@ by preventing negative stock balances.<br/>")
       (filter gnc:account-is-stock?
               (gnc-account-get-descendants-sorted
                (gnc-get-current-root-account)))
-      (list ACCT-TYPE-ASSET ACCT-TYPE-BANK ACCT-TYPE-STOCK ACCT-TYPE-MUTUAL))
+      (list ACCT-TYPE-ASSET ACCT-TYPE-STOCK ACCT-TYPE-MUTUAL))
```
Comment (dawansv) [inline gnucash/report/reports/standard/advanced-portfolio.scm]: You are accidentally reverting the last commit to advanced-portfolio.scm here accidentally. Most likely due to your struggle with rebasing to stable. https://github.com/Gnucash/gnucash/commit/aca602103e113c6e0d33ad30e8d11a0926ed5eba

**Comment by dawansv** [comment]
> Should I close the pull request and reopen it?

I am not sure as I am less familiar with the last step of the process. But as highlighted above your main commit is accidentally reverting a change that was recently merged into stable. But the rest of the code in the main commit looks fine.

@jralls What do you advise here? Do you want @fabrica64 to properly rebase, or can you work with what is here making sure that last commit doesn't get reverted?

**Comment by fabrica64** [comment]
> > Should I close the pull request and reopen it?
> 
> I am not sure as I am less familiar with the last step of the process. But as highlighted above your main commit is accidentally reverting a change that was recently merged into stable. But the rest of the code in the main commit looks fine.

Yes, it was accidentally. I synchronized the stable branch and it generated the reverting commit

 I should have forked the stable branch, created a working-branch and do the pull-request from the working branch

I can close the PR and reopen it in the right way. Let me know if this is ok

**Comment by jralls** [comment]
> I should have forked the stable branch, created a working-branch and do the pull-request from the working branch

You already forked the stable branch. All you need to do is create the new branch and push it
```
git branch money-out
git push fabrica64 money-out
```
After that you probably want to make your stable branch match origin's:
```
git checkout stable
git reset --hard origin/stable
```

> I can close the PR and reopen it in the right way. Let me know if this is ok

You can do it that way, but once you've created the new branch and pushed it to your fork you can just edit the branch on this PR to point to the new branch. The edit button is the overlapped-boxes icon at the right end of "Open
[fabrica64](https://github.com/fabrica64) wants to merge 3 commits into [Gnucash:stable](https://github.com/Gnucash/gnucash) from [fabrica64:stable](https://github.com/fabrica64/gnucash)"

After doing rebases you can force push to your feature branch (`git push -f fabrica64 money-out`) to update the PR, no need to make a new one.

**Comment by dawansv** [comment]
> you can just edit the branch on this PR to point to the new branch

Actually I don't think that's possible for the origin branch. You can change the target branch but not the origin one which is what would be needed in this case.

**Comment by dawansv** [comment]
> After that you probably want to make your stable branch match origin's:
> 
> ```
> git checkout stable
> git reset --hard origin/stable
> ```

True for you John as a core developer with write access but not us contributors. Or at least not if following the naming scheme on the wiki. @fabrica64 if you follow the instructions on the wiki, you should end up with 2 remotes:
- origin which points to your fork with right access (what John calls fabrica64)
- upstream which point to the main gnucash repository (what John calls origin)

Of course remote names are whatever you want them to be, you could use fabrica64 and origin as per John's example. So of course check what they are on your system. But if you follow the naming convention there can be some confusion between origin for core developers and origin for contributors. For core developpers origin is the main gnucash repo while for contributors origin is their fork. This command will show what what they are:

`git remote -v`

So if using the standard naming convention, to reset your stable branch to the latest upstream branch you would use the following. **But make sure you create your money-out branch first as per John's instructions otherwise you will lose your changes.**
```
git checkout stable
git reset --hard upstream/stable
```
You will still be left with that unwanted last commit on money-out. The following will let you remove it.
But first make sure it's there at the top with :

`git log`

You should see that most recent commit [Merge branch 'Gnucash:stable' into stable] at the top. The following will remove it, it essentially resets the branch to the state before the last commit. **So only runs this once!**
```
git checkout money-out
git reset --hard HEAD~1
```

Now check that it's gone:

`git log`


Then if you want you can rebase stable into money-out, meaning you bring the most recent changes from stable into money-out before your own changes, moving your own changes back to the top. But this is not the same as a new commit as you did before. It's a different process using the git rebase command.
```
git checkout money-out
git rebase stable money-out
```

If there is a conflict follow instructions to fix.

Then you can push money-out to origin; the first time you can use:
```
git push origin money-out
```

But if money-out already exist on your fork, then if you did a rebase you must use the -f flag to force the update
```
git push -f origin money-out
```

This last step is how you can keep updating your local branch with added code, etc; when the PR is created with the money-out branch, then as soon as you push your updates to origin, the PR gets updated automatically.

**Comment by fabrica64** [comment]
> > you can just edit the branch on this PR to point to the new branch
> 
> Actually I don't think that's possible for the origin branch. You can change the target branch but not the origin one which is what would be needed in this case.

I checked, it is not possible. I created another PR, #1660, this time from a branch.

**Pull request closed without merging**